Lesson4/Program.cs

```diff
@@ -1,15 +1,198 @@
 ﻿using System;
+using System.IO;
+using System.Reflection;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Json;
+using System.Xml;
 
 namespace Lesson4
 {
     class Program
     {
         static void Main(string[] args)
         {
+            var path = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            var file = "log.txt";
+
+            var configPack1 = Parser.Deserialize(path + "\\" + file);
+            if (configPack1 == null)
+                configPack1 = new ConfigPack(1, 10f, 1, "string");
+
+            var fileWatcher = new FileWatcher(path, file);
+            configPack1.Subscribe(fileWatcher);
+
+            var configuration = new Configurator(configPack1);
+            
+            fileWatcher.Start();
+
+            Console.WriteLine("Press 'q' to quit.");
+            while (Console.ReadKey().Key != ConsoleKey.Q) ;
+
+            fileWatcher.Stop();
+            configPack1.Unsubscribe(fileWatcher);
+
+        }
+    }
+
+    public interface IConfigObservable
+    {
+        void AddListener(Action<ConfigPack> action);
+        void RemoveListener(Action<ConfigPack> action);
+    }
+
+    public static class Parser
+    {
+        public static bool Serialize(string filePath, ConfigPack config)
+        {
+            bool isSerialized = true;
+            FileStream fileStream = null;
+            XmlDictionaryWriter json = null;
+
+            try
+            { 
+                var serializer = new DataContractJsonSerializer(typeof(ConfigPack));
+                fileStream = new FileStream(filePath, FileMode.Create);
+                json = JsonReaderWriterFactory.CreateJsonWriter(fileStream,
+                        Encoding.GetEncoding("utf-8"));
+                serializer.WriteObject(json, config);
+                json.Flush();
+            }
+            catch (Exception exception)
+            {
+                Console.WriteLine(exception.Message);
+                isSerialized = false;
+            }
+            finally
+            {
+                json?.Dispose();
+                fileStream?.Dispose();
+            }
+
+            return isSerialized;
```

Comment:
https://stackoverflow.com/questions/10255774/if-i-return-out-of-a-try-finally-block-in-c-sharp-does-the-code-in-the-finally-a?lq=1